src/rpc/service_eth.go

```diff
@@ -70,32 +126,27 @@ func (service *EthService) SendRawTransaction(ctx context.Context, s string) (*c
 		return nil, &EncodingError{StatusCode: -32602, Err: err}
 	}
 	tx := new(txtypes.Transaction)
+
 	if err := tx.UnmarshalBinary(b); err != nil {
 		return nil, &EncodingError{StatusCode: -32602, Err: err}
 	}
 
 	txHash := tx.Hash()
-	_, sent := service.processedTransactions[txHash]
-	if sent {
-		Logger.Info().Hex("Tx hash", txHash.Bytes()).Msg("Transaction already sequenced")
-		return &txHash, nil
-	}
-
-	signer := txtypes.NewLondonSigner(tx.ChainId())
-	fromAddress, err := signer.Sender(tx)
+	fromAddress, err := utils.SenderAddress(tx)
 	if err != nil {
 		return nil, &EncodingError{StatusCode: -32602, Err: err}
 	}
-	accountNonce, err := service.processor.Client.NonceAt(ctx, fromAddress, nil)
+
+	accountNonce, err := service.Processor.Client.NonceAt(ctx, fromAddress, nil)
```

Comment:
Shouldn't we go for PendingNonceAt here?

Reply:
I've opened another PR to look into it and test it as well: https://github.com/shutter-network/encrypting-rpc-server/issues/25